api/internal.c

```diff
@@ -130,9 +134,17 @@ void _function_callback_wrapper(mambo_context *ctx, watched_func_t *func) {
     ctx->event_type = POST_FN_C;
     func->post_callback(ctx);
 
-    emit_pop(ctx, (1 << es) | (1 << lr));
+#if defined __arm__ || __aarch64__
+      emit_pop(ctx, (1 << es) | (1 << lr));
+#elif defined __riscv
+      emit_pop(ctx, (1 << x8) | (1 << ra));
+#endif
     // IHL(LR) - emulated return to the caller of malloc()
-    emit_indirect_branch_by_spc(ctx, lr);
+#if defined __arm__ || __aarch64__
```

Comment:
We should add a warning to `emit_indirect_branch_by_spc()` and leave `_function_callback_wrapper()` unmodified.